R/dfm_compress.R

```diff
@@ -74,14 +79,18 @@ dfm_compress.dfmSparse <- function(x, margin = c("both", "documents", "features"
 #' # equivalent
 #' dfm(mydfm, groups = "grp")
 #' dfm(mydfm, groups = c(1, 1, 2, 2))
-dfm_group <- function(x, groups = NULL) {
+dfm_group <- function(x, groups = NULL, fill = FALSE) {
     UseMethod("dfm_group")
 }
 
 #' @noRd
 #' @export
-dfm_group.dfm <- function(x, groups = NULL) {
-    dfm(x, groups = groups, tolower = FALSE)
+dfm_group.dfm <- function(x, groups = NULL, fill = FALSE) {
+    #dfm(x, groups = groups, tolower = FALSE)
+    if (is.character(groups) & all(groups %in% names(docvars(x)))) {
+        groups <- interaction(docvars(x)[, groups], drop = TRUE)
```

Comment:
This is the line that automatically makes a factor of any input vectors, and that may not be desired since the documents are then re-ordered in the factor order. See tests below.